File: laba2/go/3-8.go
package main

import "fmt"

func main() {
	var n int
	fmt.Println("Введите ваше N: ")
	fmt.Scan(&n)

	numbers := []int{}
	fmt.Println("Введите ваши числа: ")
	for i := 0; i < n; i++ {
		var num int
		fmt.Scan(&num)
		numbers = append(numbers, num)
	}
	var count int
	count = 0
	for i := 1; i < len(numbers); i++ {
		if numbers[i] == numbers[i-1] {
			count++
		}
	}
	fmt.Println("Количество чисел, равных предыдущему: ", count)

}

File: laba2/swift/3.8.swift
import Foundation

var numbers: [Int] = []

print("Введите ваше N: ", terminator: "")

if let input = readLine(), let n = Int(input) {
    
    print("Введите ваши числа:")
    
    if let line = readLine() {
        let parts = line.split(separator: " ")
        
        for i in 0..<min(n, parts.count) {
            if let num = Int(parts[i]) {
                numbers.append(num)
            }
        }
    }

    var count = 0
    
    if numbers.count > 1 {
        for i in 1..<numbers.count {
            if numbers[i] == numbers[i-1] {
                count += 1
            }
        }
    }
    
    print("Количество чисел, равных предыдущему: \(count)")
}